packages/nextra/src/loader.ts

```diff
@@ -157,34 +166,73 @@ async function loader(
     unstable_flexsearch
   }
 
+  if (typeof cachedGlobalThemeStyle === 'undefined') {
+    const themeIncludeStyles = OFFICIAL_THEMES.includes(theme)
+    if (themeIncludeStyles) {
+      try {
+        const globalStylePath = await promisify(context.resolve)(
+          context.rootContext,
+          theme + '/style.css'
+        )
+        if (globalStylePath) {
+          cachedGlobalThemeStyle = readFileSync(globalStylePath).toString()
+          cachedGlobalThemeStyle = `<style jsx global>{${JSON.stringify(
+            cachedGlobalThemeStyle
+          )}}</style>`
+        } else {
+          cachedGlobalThemeStyle = null
+        }
+      } catch (e) {
+        cachedGlobalThemeStyle = null
+      }
+    } else {
+      cachedGlobalThemeStyle = null
+    }
+  }
+
+  const pageNextRoute =
+    '/' +
+    path
+      .relative(pagesDir, resourcePath)
+      // Remove the `mdx?` extension
+      .replace(MARKDOWN_EXTENSION_REGEX, '')
+      // Remove the `*/index` suffix
+      .replace(/\/index$/, '')
+      // Remove the only `index` route
+      .replace(/^index$/, '')
+
   return `
-import { withSSG as __nextra_withSSG__ } from 'nextra/ssg'
+import { SSGContext as __nextra_SSGContext__ } from 'nextra/ssg'
 import __nextra_withLayout__ from '${layout}'
 ${layoutConfig && `import __nextra_layoutConfig__ from '${layoutConfig}'`}
-${result.replace('export default MDXContent;', '')}
+
+${result}
 
 const __nextra_pageOpts__ = ${JSON.stringify(pageOpts)}
 
 globalThis.__nextra_internal__ = {
   pageMap: __nextra_pageOpts__.pageMap,
-  route: __nextra_pageOpts__.route
+  route: __nextra_pageOpts__.route,
 }
 
-const NextraLayout = __nextra_withSSG__(__nextra_withLayout__({
-  titleText: typeof titleText === 'string' ? titleText : undefined,
-  ...__nextra_pageOpts__
-}, ${layoutConfig ? '__nextra_layoutConfig__' : 'null'}))
-
-function NextraPage(props) {
-  return (
-    <NextraLayout {...props}>
+function Content(props) {
+  return <>
+    ${cachedGlobalThemeStyle || ''}
```

Comment:
Since `cachedGlobalThemeStyle` Is global style you can put it inside `__nextra_SSGContext__.Provider` and remove unneeded fragment wrapper